src/qt/listctrl.cpp

```diff
@@ -1836,14 +1836,33 @@ void wxListCtrl::CheckItem(long item, bool check)
     m_model->CheckItem(item, check);
 }
 
-void wxListCtrl::SetSingleStyle(long WXUNUSED(style), bool WXUNUSED(add))
+void wxListCtrl::SetSingleStyle(long style, bool add)
 {
+    long flag = GetWindowStyleFlag();
+
+    // Get rid of conflicting styles
+    if (add)
+    {
+        if (style & wxLC_MASK_TYPE)
+            flag = flag & ~wxLC_MASK_TYPE;
+        if (style & wxLC_MASK_ALIGN)
+            flag = flag & ~wxLC_MASK_ALIGN;
+        if (style & wxLC_MASK_SORT)
+            flag = flag & ~wxLC_MASK_SORT;
+    }
+
+    if (add)
+        flag |= style;
+    else
+        flag &= ~style;
+
+    SetWindowStyleFlag(flag);
 }
 
 void wxListCtrl::SetWindowStyleFlag(long style)
 {
     m_windowStyle = style;
-    m_qtTreeWidget->setHeaderHidden((style & wxLC_NO_HEADER) != 0);
+    m_qtTreeWidget->setHeaderHidden((style & (wxLC_NO_HEADER | wxLC_LIST)) != 0);
```

Comment:
Why only `wxLC_LIST`? We should also hide it for `wxLC_ICON` and `wxLC_SMALL_ICON`, so I'd do something like

```suggestion
    m_qtTreeWidget->setHeaderHidden((style & wxLC_NO_HEADER) != 0 || (style & wxLC_REPORT) == 0);
```